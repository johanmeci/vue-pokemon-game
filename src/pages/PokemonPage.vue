<template>
  <h1>¿Quien es este Pokémon?</h1>

  <!--Picture-->
  <PokemonPicture :pokemonId="4" :showPokemon="true" />

  <!--Options-->
  <PokemonOptions />

</template>

<script>
import PokemonPicture from '@/components/PokemonPicture'
import PokemonOptions from '@/components/PokemonOptions'

export default {
  components: {
    PokemonPicture,
    PokemonOptions
  }
}
</script>

<style>

</style>