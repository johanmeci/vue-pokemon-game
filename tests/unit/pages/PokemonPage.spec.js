import { shallowMount, mount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonPage Component', () => {

  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(PokemonPage)
  })

  test('debe llamar el mixPokemonArr al montar', () => {

    const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
    const wrapper = shallowMount(PokemonPage)
    expect(mixPokemonArraySpy).toHaveBeenCalled()

  })

  test('debe hacer match con el snapshot cuando cargan los pokemons', () => {

    //shallowMount -> montar los elementos sutilmente, sin hacer todo el ciclo de vida
    //mount -> si hace todo el proceso
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: ''
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

  })
  
})