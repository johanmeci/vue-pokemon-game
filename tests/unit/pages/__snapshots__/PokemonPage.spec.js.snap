// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>¿Quién es este Pokémon?</h1>
  <div class="pokegame-container">
    <!--Picture-->
    <pokemon-picture-stub pokemonid="5" showpokemon="false"></pokemon-picture-stub>
    <!--Options-->
    <!-- @selection-pokemon / @selectionPokemon -->
    <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  </div><!-- <template v-if="showAnswer"></template> -->
  <!--v-if-->
</div>
`;
